### PR TITLE
Link items to '/simc'

### PR DESCRIPTION
**Pull request by geheur**
This adds the ability to link items to '/simc' to have them be included in the output, similar to issue #8, but including the linked items at the end of output rather than having only the linked items in the output.

This is my first time making a pull request so please let me know if I did something incorrectly.

### CODE REVIEW

File: extras.lua [hunk 1]
```diff
@@ -190,3 +190,31 @@ Simulationcraft.upgradeTable = {
   [531] = 2 -- 2/2 -> 10
 }
 
+Simulationcraft.slotIds = {
```
Comment (seriallos) [inline extras.lua]: I think the slot map is a little off. Tried linking a weapon and it maps it to a trinket slot in the simc output.

This is what I have in some Raidbots code for slotId -> simc slot name that I think maps to the `INV_` slot names:

```
  export const slotMap = {
    1: 'head',
    2: 'neck',
    3: 'shoulder',
    4: 'shirt',
    5: 'chest',
    6: 'waist',
    7: 'legs',
    8: 'feet',
    9: 'wrist',
    10: 'hands',
    11: 'finger',
    12: 'trinket',
    13: 'main_hand', // weapon
    14: 'off_hand', // shield
    15: 'main_hand',
    16: 'back',
    17: 'main_hand', // two-hand
    19: 'tabard',
    20: 'chest',
    21: 'main_hand',
    22: 'off_hand',
    23: 'off_hand',
    26: 'main_hand', // some ranged weapons
    28: 'off_hand',
  };
```

File: core.lua [hunk 2]
```diff
@@ -89,14 +89,51 @@ function Simulationcraft:UpdateMinimapButton()
   end
 end
 
+local function splitLinks(links)
```
Comment (seriallos) [inline core.lua]: Simpler version using regexes that might be a bit more robust:

```
local function splitLinks(links)
  local separatedLinks = {}
  for link in links:gmatch("(|c.-|h|r)") do
    separatedLinks[#separatedLinks + 1] = link
  end
  return separatedLinks
end
```

Looks for the `|c STUFF |h|r` patterns and adds them to the table.

File: core.lua [hunk 3]
```diff
@@ -89,14 +89,51 @@ function Simulationcraft:UpdateMinimapButton()
...
+  end
+  return separatedLinks
+end
+
 function Simulationcraft:HandleChatCommand(input)
   local args = {strsplit(' ', input)}
 
   local debugOutput = false
   local noBags = false
+  local links = {}
```
Comment (seriallos) [inline core.lua]: With the above function, the link detection code can be way simpler.

Just need to have

```
local links = splitLinks(input)
```

And it works out. The rest of the arg parsing will not match any of the link fragments but still catch any of the keywords.

**Comment by seriallos** [comment]
Make the changes requested above for the simpler item parsing and then make sure the slot map is updated and then I'll do another test pass, merge it in, and prep a release!

Nice work! Thanks for putting this together.

**Comment by geheur** [comment]
Sorry for the wait - wanted to double check some stuff related to the slot numbering. The original INV_ to slot id table I had was copied off of wowwiki and untested - however, I have confirmed that it was correct. However, the simc addon does not use the slot ids, but instead uses its own numbering system. I've updated the table to have the correct values. These values are also different from those you posted - I do not know what those correspond to.

I've also updated the link parsing to use regex; that was much cleaner, thanks.

File: extras.lua (hunk 1 above)
Comment (geheur) [reply extras.lua]: The original INV_ to slot id table I had was copied off of wowwiki and untested - however, I have confirmed that it was correct. However, the simc addon does not use the slot ids, but instead uses its own numbering system. I've updated the table to have the correct values. These values are also different from those you posted - I do not know what those correspond to.

File: core.lua (hunk 2 above)
Comment (geheur) [reply core.lua]: Done!

File: core.lua (hunk 3 above)
Comment (geheur) [reply core.lua]: Done!

**Comment by seriallos** [comment]
Sorry for my own delay! Adding some documentation and then will make a release today.

**Pull request merged**